citestwheel/citestwheel.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+set -eoxu pipefail
+
+export RAPIDS_PY_WHEEL_NAME="${RAPIDS_PY_WHEEL_NAME:-}"
+export RAPIDS_WHEEL_VER_OVERRIDE="${RAPIDS_WHEEL_VER_OVERRIDE:-}"
+export CIBW_TEST_EXTRAS="${CIBW_TEST_EXTRAS:-}"
+export CIBW_TEST_REQUIRES="${CIBW_TEST_REQUIRES:-}"
+export CIBW_TEST_COMMAND="${CIBW_TEST_COMMAND:-}"
+export RAPIDS_BEFORE_TEST_COMMANDS_AMD64="${RAPIDS_BEFORE_TEST_COMMANDS_AMD64:-}"
+export RAPIDS_BEFORE_TEST_COMMANDS_ARM64="${RAPIDS_BEFORE_TEST_COMMANDS_ARM64:-}"
+
+rm -rf ./dist
+mkdir -p ./dist
+
+first='yes'
+arch=$(uname -m)
+
+for pyver in ${RAPIDS_PY_VERSION}; do
+        deactivate || true
+
+        pytestenv="cp${pyver//./}-cp${pyver//./}"
+
+        /opt/python/$pytestenv/bin/python -m venv /cibw-test-venv-${pyver}
+        . /cibw-test-venv-${pyver}/bin/activate
+
+        curl -sS https://bootstrap.pypa.io/get-pip.py | python3
+
+        if [ "${arch}" == "x86_64" ]; then
+                sh -c "${RAPIDS_BEFORE_TEST_COMMANDS_AMD64}"
```

Comment:
This serves two (not really related) purposes:
* Shell/OS commands for pre-test setup
* Installing inter-rapids dependencies i.e. cudf-cu11 installs rmm-cu11 from CI pypi index

Installing inter-rapids -cu11 dependencies should be its own parameter:
`internal-requires`, for example?

Reply:
Internal-requires should be accurately reflected in the setup.py + extra-index-url - installing cudf-cu11 should suffice to install rmm-cu11 from the stage/nightly index.